app/session-manager.coffee

```diff
@@ -1,27 +1,34 @@
 warp = require 'nexus-warp'
 
+User = require './models/User'
 SandboxFacet     = require './facets/sandbox/sandbox-facet'
-PlayerFacet      = require './facets/player/player-facet'
-GameMasterFacet  = require './facets/game-master/game-master-facet'
+ParticipantFacet = require './facets/participant/participant-facet'
 
 class SessionManager
 
 	constructor: (@service) ->
 
-	create: (transport, cookies) ->
-		auth = cookies.find ({name}) -> name is 'auth'
-
-		facet =
-			if auth?
-				switch auth.value
-					when 'player' then new PlayerFacet @service
-					when 'game_master' then new GameMasterFacet @service
-			else
-				new SandboxFacet @service.sandbox
-
-		new warp.Session
-			facet:     facet
-			transport: transport
-			entities:  facet.entities
+	create: (transport, cookies, done) ->
+		user = {}
+
+		session_cookie = cookies.find ({name}) ->
+			name is 'koa:sess'
+
+		passport_data = JSON.parse (new Buffer session_cookie.value, 'base64').toString 'utf8'
```

Comment:
koa:sess cookie is not just passport data, but whole session, so variable name is bit confusing